### PR TITLE
cabal-docspec and hlint

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: .hlint.yaml [hunk 1]
```diff
@@ -8,3 +8,7 @@
 
 # this doesn't work well with Rank2Types
 - ignore: {name: Eta reduce}
+- ignore: {name: Unused LANGUAGE pragma}
```
Comment (RyanGlScott) [inline .hlint.yaml:11]: Have you checked to see if the suggestions that `hlint` provides here are reasonable? I'm not necessarily opposed to ignoring certain `hlint` warnings, but if the warnings are reasonable, I'd prefer just applying the suggestions.

Comment (phadej) [reply .hlint.yaml:11]: Fair. I cleaned up the other warnings but left `LANGUAGE` one. It has false positives (`DeriveTraversable` not always implies `DeriveFunctor`, e.g.) so I just left it.

**Comment by phadej** [comment]
The build is green except notification sending failed

**Comment by RyanGlScott** [comment]
> [< :beckett.freenode.net 433 * github-actions :Nickname is already in use.](https://github.com/ekmett/ad/pull/90/checks?check_run_id=1677961037#step:2:24)

Urk, it looks like this conflicted with another concurrent use of the nick `github-actions`. I wonder if this limitation is inherent in the way these sorts of IRC notification bots work, or if it would be worth investigating another IRC action (e.g., [`notify-irc`](https://github.com/marketplace/actions/notify-irc)).

In any case, thanks. Let's land this.

**Comment by phadej** [comment]
I think `notify-irc` has the same problem, i.e. everything which connects on run is not gonna work.

I'm half-seriously thinking of making an IRC bot just for these, which would keep persistent connection.

**Comment by RyanGlScott** [comment]
> I'm half-seriously thinking of making an IRC bot just for these, which would keep persistent connection.

That's certainly one way to solve the problem. As an alternative, we could generate unique nicks for each notification, but perhaps that's overkill.

**Pull request merged**